pdns/pdns_recursor.cc

```diff
@@ -2170,6 +2173,8 @@ static void startDoResolve(void *p)
     }
 #endif /* NOD_ENABLED */
 
+    t_taskQueue.runOnce();
```

Comment:
Perhaps we should run that outside of this exception block? I know we are not supposed to bubble up that far but if we ever do we might be confused about seeing the context of the initial query in there.